main/main.cpp

```diff
@@ -29,38 +33,59 @@ esp_err_t set_adc(
 	adc_channel_t  adc_channel
 );
 
-QueueHandle_t refer_speed_q = xQueueCreate(2, sizeof(float));
-QueueHandle_t motor_speed_q = xQueueCreate(2, sizeof(float));
-QueueHandle_t motor_count_q = xQueueCreate(2, sizeof(int));
+QueueHandle_t refer_speed_q = xQueueCreate(1, sizeof(float));
+QueueHandle_t motor_speed_q = xQueueCreate(1, sizeof(float));
+QueueHandle_t motor_count_q = xQueueCreate(1, sizeof(int));
+
+QueueHandle_t tel_ref_speed_q   = xQueueCreate(1, sizeof(float));
+QueueHandle_t tel_motor_speed_q = xQueueCreate(1, sizeof(float));
+QueueHandle_t tel_error_q       = xQueueCreate(1, sizeof(float));
+QueueHandle_t tel_error_der_q   = xQueueCreate(1, sizeof(float));
+QueueHandle_t tel_control_signal_q = xQueueCreate(1, sizeof(float));
+QueueHandle_t tel_exec_time_q      = xQueueCreate(1, sizeof(float));
 
 typedef struct {
 	adc_oneshot_unit_handle_t adc_handle;
-	// volatile uint32_t *motor_counter;
 } timer_args;
 
 void IRAM_ATTR send_status(void* argp) {
 	timer_args *args = (timer_args*)argp;
 	adc_oneshot_unit_handle_t adc_handle = args->adc_handle;
 	const float REF_MIN = 50.f*(2*M_PI/60), REF_MAX = 400.0f*(2*M_PI/60);
-	const int N_ENCODER_SLITS = 20;
 
 	int adc_read = 0;
 	(void)adc_oneshot_read(adc_handle, ADC_CHANNEL_0, &adc_read);
 
 	BaseType_t xHigherPriorityTaskWoken = pdFALSE;
 	float adc_value = (float)adc_read*(REF_MAX-REF_MIN)/(float)(0b111111111) + REF_MIN;
 
-	float refer_speed = adc_value;
+	static int sine_lut_idx = 0;
+	static int sine_lut_idx_step = 1;
+	static float sine_lut_mult = 1.0f;
+	
+	float sine = sine_lut_mult*AMPLITUDE*sine_wave_90_deg_LUT[sine_lut_idx];
+	sine_lut_idx += sine_lut_idx_step;
+
+	if (sine_lut_idx >= QUARTER_TABLE_SIZE) {
+		sine_lut_idx_step = -1;
+		sine_lut_idx = QUARTER_TABLE_SIZE - 2;
+	}
+	if (sine_lut_idx <= 0) {
+		sine_lut_idx_step = 1;
+		sine_lut_mult *= -1.0;
+		sine_lut_idx = 0;
+	}
+	float refer_speed = adc_value + sine;
+	static float prev_motor_speed = 0.0;
 	float motor_speed = 0.0;
 
-	// 20 ranuras encoder, 20 intr por revolución
 	static int32_t motor_count = 0;
 	int32_t prev_count = motor_count;
 	xQueueReceiveFromISR(motor_count_q, &motor_count, &xHigherPriorityTaskWoken);
-	motor_speed = (float)(motor_count-prev_count)*2.0f*M_PI / (N_ENCODER_SLITS*SAMPLE_TIME_s);
+	motor_speed = (float)(motor_count-prev_count)*2.0f*M_PI / (ENCODER_SLITS*SAMPLE_TIME_s);
 
-	xQueueSendFromISR(refer_speed_q, &refer_speed, &xHigherPriorityTaskWoken);
-	xQueueSendFromISR(motor_speed_q, &motor_speed, &xHigherPriorityTaskWoken);
+	xQueueOverwriteFromISR(refer_speed_q, &refer_speed, &xHigherPriorityTaskWoken);
```

Comment:
instead of sending two values with two queues, create an struct and only send on single time.